connector/datadogconnector/connector_native.go

```diff
@@ -0,0 +1,154 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package datadogconnector // import "github.com/open-telemetry/opentelemetry-collector-contrib/connector/datadogconnector"
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	pb "github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace"
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/stats"
+	"github.com/DataDog/datadog-go/v5/statsd"
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/attributes"
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/metrics"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.opentelemetry.io/otel/metric/noop"
+	"go.uber.org/zap"
+)
+
+// traceToMetricConnectorNative is the schema for connector
+type traceToMetricConnectorNative struct {
+	metricsConsumer consumer.Metrics // the next component in the pipeline to ingest metrics after connector
+	logger          *zap.Logger
+
+	// concentrator ingests spans and produces APM stats
+	concentrator *stats.Concentrator
+
+	// tcfg is the trace agent config
+	tcfg *config.AgentConfig
+
+	// ctagKeys are container tag keys
+	ctagKeys []string
+
+	// translator specifies the translator used to transform APM Stats Payloads
+	// from the agent to OTLP Metrics.
+	translator *metrics.Translator
+
+	// in specifies the channel through which the agent will output Stats Payloads
+	// resulting from ingested traces.
+	in chan *pb.StatsPayload
+
+	// exit specifies the exit channel, which will be closed upon shutdown.
+	exit chan struct{}
+
+	// isStarted tracks whether Start() has been called.
+	isStarted bool
+}
+
+var _ component.Component = (*traceToMetricConnectorNative)(nil) // testing that the connectorImp properly implements the type Component interface
+
+// newTraceToMetricConnectorNative creates a new connector with native OTel span ingestion
+func newTraceToMetricConnectorNative(set component.TelemetrySettings, cfg component.Config, metricsConsumer consumer.Metrics, metricsClient statsd.ClientInterface) (*traceToMetricConnectorNative, error) {
+	set.Logger.Info("Building datadog connector for traces to metrics")
+	in := make(chan *pb.StatsPayload, 100)
+	set.MeterProvider = noop.NewMeterProvider() // disable metrics for the connector
```

Comment:
why are we doing it ?

Reply:
This is a copy of https://github.com/open-telemetry/opentelemetry-collector-contrib/blame/main/connector/datadogconnector/connector.go#L69 and was added in https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/9a34aea049669c76d716885d347c96c5f6e3e3dc#diff-63ca951f32af68231d95ef502d34657b930abb3d1a6ffaa8db864a28f3a17d6aR51

cc @mx-psi should have more context

Reply:
As the code stands, if we enable this, we may have duplicate metrics from the connector and exporter with the exact same name and tags. We can add support for this, but we need to fix that first.